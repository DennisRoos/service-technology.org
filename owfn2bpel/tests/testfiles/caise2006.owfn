PLACE
INTERNAL p1, p2, p3, p4, p5, p6, p7, p8, p9, p10, p11, p12, p13;
INPUT registration, quest_in;
OUTPUT quest_out, result, archive;

INITIALMARKING p1: 1;
FINALMARKING p13: 1;

TRANSITION a1
CONSUME p1, registration;
PRODUCE p2, p3;

TRANSITION a2
CONSUME p2;
PRODUCE p4, quest_out;

TRANSITION a3
CONSUME p4, quest_in;
PRODUCE p5;

TRANSITION a4
CONSUME p5;
PRODUCE p6;

TRANSITION a5
CONSUME p3;
PRODUCE p7;

TRANSITION a6
CONSUME p7;
PRODUCE p8;

TRANSITION a7
CONSUME p8;
PRODUCE p9;

TRANSITION NOK
CONSUME p9;
PRODUCE p3;

TRANSITION OK
CONSUME p9;
PRODUCE p10;

TRANSITION DONE
CONSUME p8;
PRODUCE p10;

TRANSITION a8
CONSUME p10;
PRODUCE p11, result;

TRANSITION JOIN
CONSUME p11, p6;
PRODUCE p12;

TRANSITION a9
CONSUME p12;
PRODUCE p13, archive;
