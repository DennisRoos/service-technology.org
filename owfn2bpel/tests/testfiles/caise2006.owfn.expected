<process name="Act_process_caise2006.owfn"
  targetNamespace="http://docs.oasis-open.org/wsbpel/2.0/process/abstract"
  xmlns="http://docs.oasis-open.org/wsbpel/2.0/process/abstract"
  suppressJoinFailure="yes"
  xmlns:template="http://docs.oasisopen.org/wsbpel/2.0/process/abstract/simple-template/2006/08"
  abstractProcessProfile="http://docs.oasisopen.org/wsbpel/2.0/process/abstract/simple-template/2006/08">
    <partnerLinks>
      <partnerLink name="generic_pl" partnerLinkType="##opaque"
                   myRole="##opaque" partnerRole="##opaque" />
    </partnerLinks>

  <variables>
    <variable name="Var_registration" element="##opaque" />
    <variable name="Var_quest_in" element="##opaque" />
    <variable name="Var_quest_out" element="##opaque" />
    <variable name="Var_result" element="##opaque" />
    <variable name="Var_archive" element="##opaque" />
  </variables>

  <sequence name="Act_sequence">
    <receive name="Act_registration"
      partnerLink="generic_pl"
      operation="registration"
      variable="Var_registration" />
    <flow name="Act_a1">
      <sequence name="Act_sequence">
        <invoke name="Act_quest_out"
          partnerLink="generic_pl"
          operation="quest_out"
          inputVariable="Var_quest_out" />
        <receive name="Act_quest_in"
          partnerLink="generic_pl"
          operation="quest_in"
          variable="Var_quest_in" />
        <opaqueActivity name="Act_a4" />
      </sequence>

      <sequence name="Act_sequence">
        <opaqueActivity name="Act_a5" />
        <opaqueActivity name="Act_a6" />
        <while name="Act_p3_a5_p7_a6_p8">
          <condition opaque="yes" />
          <sequence name="Act_sequence">
            <opaqueActivity name="Act_a7" />
            <opaqueActivity name="Act_NOK" />
            <opaqueActivity name="Act_a5" />
            <opaqueActivity name="Act_a6" />
          </sequence>

        </while>

        <if name="Act_if_p3_a5_p7_a6_p8_to_p10_a8_p11">
          <condition opaque="yes" />
            <sequence name="Act_sequence">
              <opaqueActivity name="Act_a7" />
              <opaqueActivity name="Act_OK" />
            </sequence>

          <else>
              <opaqueActivity name="Act_DONE" />
          </else>
        </if>

        <invoke name="Act_result"
          partnerLink="generic_pl"
          operation="result"
          inputVariable="Var_result" />
      </sequence>

    </flow>

    <opaqueActivity name="Act_JOIN" />
    <invoke name="Act_archive"
      partnerLink="generic_pl"
      operation="archive"
      inputVariable="Var_archive" />
  </sequence>

</process>

